addons/web/static/src/public/caps_lock_warning.js

```diff
@@ -0,0 +1,32 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+export class CapsLockWarning extends Interaction {
+    static selector = ".caps-lock-warning";
+    dynamicContent = {
+        ".password-input": {
+            "t-on-keydown": (ev) => this._onInputKeyDown(ev),
+        },
+    };
+
+    /**
+     * Capture keydown events to detect the CAPS LOCK state and toggle the
+     * CAPS LOCK warning accordingly
+     *
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    _onInputKeyDown(ev) {
+        // FALSE when we first hit the CAPS-LOCK
+        // at this point, the CAPS-LOCK is yet to TURN ON.
+        const state = ev.getModifierState?.("CapsLock");
+        const shouldHideWarning = ev.key === "CapsLock" ? state : !state;
+
+        // FALSE value REMOVES the `invisible` class while TRUE ADDS it.
+        this.el
+            .querySelector(".caps-lock-warning-text")
+            .classList.toggle("invisible", shouldHideWarning);
```

Comment:
To be even more inline with the new interaction system, you should declare that class addition/removal in the `dynamicContent` object (same as the t-on-keydown you added). (You'll find many examples with `"t-att-class":`).

It would actually be important in some case (e.g. Interaction which are active in edit mode).

Note: you may need to use `renderAt` in the `start` to render the element in the first place (as I asked for it not to be in the DOM during page loading).